.github/workflows/test.yml

```diff
@@ -1,34 +1,24 @@
 name: test
 
-on: pull_request
+on:
+  pull_request:  # any pull request
+  schedule:  # run weekly
+    - cron: "0 12 * * 0"
```

Comment:
what is the intuition behind running weekly, as well as on PRs? just forward-looking?

Reply:
It's an easy, convenient way to ensure code doesn't break while you're not actively looking at it, e.g. because dependent packages make breaking changes in a patch update.